### PR TITLE
Notify session about errors in ConnectionContext

### PR DESCRIPTION
**Pull request by mfussenegger**
This fixes `testErrorRecoveryFromErrorsOutsideSqlOperations` which was
flaky.

### CODE REVIEW

File: sql/src/main/java/io/crate/protocols/postgres/Messages.java [hunk 1]
```diff
@@ -190,7 +190,8 @@ public void operationComplete(ChannelFuture future) throws Exception {
      * See https://www.postgresql.org/docs/9.2/static/protocol-error-fields.html for a list of error codes
      *
      */
-    static void sendErrorResponse(Channel channel, @Nonnull final String message) {
+    static void sendErrorResponse(Channel channel, Throwable throwable) {
```
Comment (matriv) [inline sql/src/main/java/io/crate/protocols/postgres/Messages.java]: Why did you remove the @Nonnnul?

Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/Messages.java]: Could be Null now because `Exceptions.messageOf` handles null values

**Comment by matriv** [comment]
Igtm
